ARC-873 adding better error message for ip allowlist blocking (#961)
atlassian/github-for-jira: docs/ip-allowlist.md

# GitHub IP Allow List Configuration

> If your organization is using [GitHub's Organization IP Allow List](https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-allowed-ip-addresses-for-your-organization), it needs to be configured properly for this Github app to be able to communicate with your organization's GitHub API.

Reviewer: nit: Github -> GitHub

There are 2 methods to remedy this based on your company's security policy:

1. a simple method which will enable _all_ GitHub apps with IPs specified in them to have access to your GitHub org's
   APIs (_recommended_)
2. a manual way by adding each CIDR ranges possible for this app to communicate from.

**We recommend using the first method** as you only have to set it once and never have to think about it again. If there are
any IP changes on our end in the future, we can just change the list on our end and it will automatically propagate to

> your organization. But for this to happen, you must trust every GitHub app installed or else risk a potential security

Reviewer: I find it a bit awkward tbh to recommend something that can lead to a potential security risk...

Author: Installing _any app_ is a potential security risk.  This is why they need to assess that risk and see if it's worth it for the convenience.

breach by an app adding an attacker's IP to your allow list.

> If you'd like complete control over your IP Allow List, then you can enter the CIDR ranges manually in your GitHub 

Reviewer: nit: "If you'd like complete control"  ->  "If you'd like to have a complete control"?

> organization.  But it does come with the drawback that if the CIDR ranges ever change or a new one needs to be added,

Reviewer: Wondering if we have checked with micros team if we could have a fixed set of IPs? Or maybe we should use outbound proxy for GitHub calls? Is it roadmapped?

Author: We are using outbound proxy.  But even outbound proxy might add new regions and change the IP list slightly.

> you will have to manually update those as well.  Furthermore, we don't have a way to easily send a message 

Reviewer: supernit: two spaces after '.'

to all GitHub org admins about a change like this and it could be possible that the integration might break because 
of the change.

### Simple Method

As an admin go to your GitHub org page `https://github.com/<your org>`, press on the `Settings` tab, then in the 
sidebar select the `Organization security` option.  Scroll down to the `IP allow list` section.  Both checkboxes `Enable IP 
allow list` and `Enable IP allow list configuration for installed GitHub Apps` should be selected and saved independently.

![](images/github-ip-allowlist.png)

That's it!

### Manual Method

As an admin your GitHub org page `https://github.com/<your org>`, press on the `Settings` tab, then in the sidebar
select the `Organization security` option.  Scroll down to the `IP allow list` section until you can see the list of 
IP addresses with a `+ Add` button.  From here, you need to [add the whole list of CIDR ranges specified in this 
Atlassian 
document](https://support.atlassian.com/organization-administration/docs/ip-addresses-and-domains-for-atlassian-cloud
-products/#AtlassiancloudIPrangesanddomains-OutgoingConnections).

> For simplicity, here's the list of CIDR ranges, but it might not be up to date:

Reviewer: Any chance we could give them some link (e.g. https://ip-ranges.atlassian.com/) where they could find the up-do-date information?

Author: that support link has up to date information, and we probably don't want them to add every single IP in that ip-ranges list.  Plus, all that information is in the page linked.  Just the shortlist is fine.

# GitHub IP Allow List Configuration

If your organization is using [GitHub's Organization IP Allow List](https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-allowed-ip-addresses-for-your-organization)

> , it might not be configured properly for this Github app to be able to communicate with your organization's GitHub API.

Reviewer 2: ```suggestion
, it needs to be configured properly for this Github app to be able to communicate with your organization's GitHub API.
```
